cdap-docs/developers-manual/source/building-blocks/transaction-system.rst

```diff
@@ -81,30 +81,51 @@ Here are some rules to follow for Flows, Flowlets, Services, and Procedures:
 Keeping these guidelines in mind will help you write more efficient and faster-performing
 code.
 
+Levels of Conflict Detection
+----------------------------
 
-The Need for Disabling Transactions
------------------------------------
 Transactions providing ACID (atomicity, consistency, isolation, and durability) guarantees
 are useful in several applications where data accuracy is critical—examples include billing
 applications and computing click-through rates.
 
-However, some applications—such as trending—might not need it. Applications that do not
-strictly require accuracy can trade off accuracy against increased throughput by taking
-advantage of not having to write/read all the data in a transaction.
-
-Disabling Transactions
-----------------------
-Transactions can be disabled for a Flow by annotating the Flow class with the
-``@DisableTransaction`` annotation::
-
-  @DisableTransaction
-  class MyExampleFlow implements Flow {
-    ...
-  }
-
-While this may speed up performance, if—for example—a Flowlet fails, the system would not
-be able to roll back to its previous state. You will need to judge whether the increase in
-performance offsets the increased risk of inaccurate data.
+However, transaction are not for free: The transaction system must track all the writes
+made by all transactions, and it must check transactions for conflicts before committing them.
+Also, if conflicts are frequent, then they impact performance, because the failed transactions
+have to be rolled back and reattempted.
```

Comment:
However, transaction are not for free: the transaction system must track all the writes
made by all transactions, and it must check transactions for conflicts before committing them.
If conflicts are frequent, they will impact performance because the failed transactions
have to be rolled back and reattempted.